geoutils/raster/distributed_computing/multiproc.py

```diff
@@ -0,0 +1,194 @@
+# Copyright (c) 2025 Centre National d'Etudes Spatiales (CNES)
+#
+# This file is part of the GeoUtils project:
+# https://github.com/glaciohack/geoutils
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+#
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Process out-of-memory calculations"""
+from typing import Any, Callable
+
+import numpy as np
+import rasterio as rio
+
+from geoutils import Raster
+from geoutils._typing import NDArrayNum
+from geoutils.raster import RasterType, compute_tiling
+from geoutils.raster.distributed_computing.cluster import (
+    AbstractCluster,
+    ClusterGenerator,
+)
+
+
+def load_raster_tile(raster_unload: RasterType, tile: NDArrayNum) -> RasterType:
+    """
+    Extracts a specific tile (spatial subset) from the raster based on the provided tile coordinates.
+
+    :param raster_unload: The input raster from which the tile is to be extracted.
+    :param tile: The bounding box of the tile as [xmin, xmax, ymin, ymax].
+    :return: The extracted raster tile.
+    """
+    xmin, xmax, ymin, ymax = tile
+    # Crop the raster to extract the tile based on the bounding box coordinates
+    raster_tile = raster_unload.icrop(bbox=(xmin, ymin, xmax, ymax))
+    return raster_tile
+
+
+def remove_tile_padding(raster: RasterType, raster_tile: RasterType, tile: NDArrayNum, padding: int) -> None:
+    """
+    Removes the padding added around tiles during terrain attribute computation to prevent edge effects.
+
+    :param raster: The full DEM object from which tiles are extracted.
+    :param raster_tile: The raster tile with possible padding that needs removal.
+    :param tile: The bounding box of the tile as [xmin, xmax, ymin, ymax].
+    :param padding: The padding size to be removed from each side of the tile.
+    """
+    # New bounding box dimensions after removing padding
+    xmin, xmax, ymin, ymax = 0, raster_tile.height, 0, raster_tile.width
+
+    # Remove padding only if the tile is not at the DEM's edges
+    if tile[0] != 0:
+        tile[0] += padding
+        xmin += padding
+    if tile[1] != raster.height:
+        tile[1] -= padding
+        xmax -= padding
+    if tile[2] != 0:
+        tile[2] += padding
+        ymin += padding
+    if tile[3] != raster.width:
+        tile[3] -= padding
+        ymax -= padding
+
+    # Apply the new bounding box to crop the tile and remove the padding
+    raster_tile.icrop(bbox=(xmin, ymin, xmax, ymax), inplace=True)
+
+
+def map_block(
```

Comment:
We could name it `apply_func_block` here instead (`map` is usually for the function that distributes the function to all blocks).